StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1129CodeFixProvider.cs

```diff
@@ -49,21 +50,116 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        /// </summary>
+        /// <typeparam name="T">The type to match.</typeparam>
+        /// <param name="namedTypeSymbol">The symbol.</param>
+        /// <returns><see langword="true"/> if the syntax matches the type; <see langword="false"/> otherwise.</returns>
+        private static bool IsType<T>(INamedTypeSymbol namedTypeSymbol)
+        {
+            if (namedTypeSymbol == null)
+            {
+                return false;
+            }
+
+            var expectedType = typeof(T);
+
+            if (!string.Equals(expectedType.Name, namedTypeSymbol.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            if (!string.Equals(
+                expectedType.Namespace,
+                namedTypeSymbol.ContainingNamespace?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted)),
+                StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether a given enumeration symbol contains a member with value <c>0</c>.
+        /// </summary>
+        /// <param name="namedTypeSymbol">The symbol.</param>
+        /// <param name="foundMemberName">Will be set to the string name of the member, if one is found.</param>
+        /// <returns><see langword="true"/> if the syntax is an enumeration with a value of <c>0</c>; <see langword="false"/> otherwise.</returns>
+        private static bool IsEnumWithDefaultMember(INamedTypeSymbol namedTypeSymbol, out string foundMemberName)
+        {
+            foundMemberName = null;
+
+            if (namedTypeSymbol == null || namedTypeSymbol.TypeKind != TypeKind.Enum)
+            {
+                return false;
+            }
+
+            var foundMembers = namedTypeSymbol
+                .GetMembers()
+                .Where(m => m.Kind == SymbolKind.Field)
+                .OfType<IFieldSymbol>()
+                .Where(fs => fs.ConstantValue.Equals(0))
+                .ToList();
+
+            if (foundMembers.Count != 1)
+            {
+                return false;
+            }
+
+            foundMemberName = foundMembers.Single().Name;
+            return true;
+        }
+
+        /// <summary>
+        /// Gets a qualified member access expression for the given <paramref name="typeSyntax"/>.
+        /// </summary>
+        /// <param name="typeSyntax">The type syntax from the original constructor.</param>
+        /// <param name="memberName">The member name.</param>
+        /// <returns>A new member access expression.</returns>
+        private static SyntaxNode ConstructMemberAccessSyntax(TypeSyntax typeSyntax, string memberName)
+            => SyntaxFactory.MemberAccessExpression(
+                SyntaxKind.SimpleMemberAccessExpression,
+                typeSyntax,
+                SyntaxFactory.IdentifierName(memberName));
```

Comment:
I personally wouldn't use the arrow syntax if the expression spans multiple lines. But I don't think it is a big deal.